### PR TITLE
Add task-specific YOLO26 training hyperparameters for pose and OBB

### PR DESCRIPTION
**Pull request by glenn-jocher**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary

Added task-specific training hyperparameter overrides for pose and OBB models while updating the shared YOLO26 training defaults and documentation to the new tuned values.

### 📊 Key Changes

- Added `TASK_HYPS` configurations in `ultralytics/cfg/__init__.py` for `pose` and `obb`, including task-specific learning rates, loss weights, warmup settings, and augmentations.
- Applied these task-specific values during training when the corresponding settings remain at their `default.yaml` values; explicit user overrides are merged afterward and take precedence.
- Updated `ultralytics/cfg/default.yaml` with new default optimization, loss, and augmentation hyperparameters.
- Updated the training argument documentation to reflect the changed shared defaults.

### 🎯 Purpose & Impact

- Pose and OBB training now use different built-in hyperparameters, including stronger pose loss and keypoint objectness weights for pose, and 180-degree rotation augmentation for OBB.
- Training configurations that rely on shared defaults will use the new values; users specifying hyperparameters explicitly retain those settings.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help streamline review; an engineer will assist soon. Please review the checklist below to support a smooth integration of your YOLO26 hyperparameter fix 🛠️✨

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/cfg/__init__.py [hunk 1]
```diff
@@ -307,6 +371,13 @@ def get_cfg(
     if overrides:
         overrides = cfg2dict(overrides)
         check_dict_alignment(cfg, overrides)
+
+        # Apply task-specific training hyperparameters between defaults and user overrides
+        if overrides.get("mode", cfg.get("mode", "")) == "train":
+            task_hyps = TASK_HYPS.get(overrides.get("task", cfg.get("task", "")), {})
+            if task_hyps:
+                cfg = {**cfg, **task_hyps}
```
Comment (UltralyticsAssistant) [inline ultralytics/cfg/__init__.py]: ⚠️ **HIGH**: `cfg` here is already the fully loaded base config, which can be a user-provided YAML rather than the repo's `default.yaml`. Merging `TASK_HYPS` at this point will silently overwrite explicit pose/obb hyperparameters from custom config files unless the user repeats them in `overrides`, which is a behavior regression. Please apply these task defaults only to the stock defaults path, or only for keys that were not explicitly set by the incoming cfg.

File: ultralytics/cfg/default.yaml [hunk 2]
```diff
@@ -92,36 +92,36 @@ workspace: # (float, optional) engine (TensorRT) only; workspace size in GiB, e.
...
+perspective: 0.00074 # (float) perspective fraction (0–0.001 typical)
+flipud: 0.00653 # (float) vertical flip probability
+fliplr: 0.30393 # (float) horizontal flip probability
 bgr: 0.0 # (float) RGB↔BGR channel swap probability
-mosaic: 1.0 # (float) mosaic augmentation probability
-mixup: 0.0 # (float) MixUp augmentation probability
-cutmix: 0.0 # (float) CutMix augmentation probability
-copy_paste: 0.0 # (float) segmentation copy-paste probability
+mosaic: 0.99182 # (float) mosaic augmentation probability
+mixup: 0.42713 # (float) MixUp augmentation probability
```
Comment (UltralyticsAssistant) [inline ultralytics/cfg/default.yaml:128]: 💡 **MEDIUM**: `default.yaml` is shared across tasks, including classification. `mixup`/`cutmix` are used by classify training, so changing them to detect/segment-tuned values here alters classification behavior too, but `TASK_HYPS` only restores pose and OBB. If these numbers are intended only for detect/segment, classify needs its own override or task-gated defaults to avoid an unrelated regression.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The task-specific override approach makes sense, but there are two behavior regressions to address before merging: the new merge order can clobber values from user-supplied pose/OBB config files, and the shared `default.yaml` changes also affect classification training without any compensating task-specific override.

💬 Posted 2 inline comments

**Comment by Laughing-q** [comment]
@glenn-jocher before moving forward on this please check message on slack boss
